extensions-contrib/opentelemetry-emitter/README.md

```diff
@@ -0,0 +1,170 @@
+# OpenTracing Emitter
+
+The OpenTelemetry emitter generates OpenTelemetry Spans for queries.
+
+
+## Configuration
+
+### Enabling
+
+Load the plugin and enable the emitter in `common.runtime.properties`:
+
+Load the plugin:
+
+```
+druid.extensions.loadList=[..., "opentelemetry-emitter"]
+```
+
+Enable the emitter:
+
+```
+druid.emitter=opentelemetry
+```
+
+*_Don't forget to remove the line_ `druid.emitter=noop` _to avoid rewriting_.
+
+## Testing
+###Zipkin
+Run zipkin docker container
+```
+docker run -d -p 9411:9411 openzipkin/zipkin
+```
+Zipkin will be available at `http://localhost:9411`.
+
+Add zipkin dependency to `extensions-contrib/opentelemetry-emitter/pom.xml`:
+```
+<dependency>
+  <groupId>io.opentelemetry</groupId>
+  <artifactId>opentelemetry-exporter-zipkin</artifactId>
+  <version>${your.version}</version>
+</dependency>
+```
+
+Run:
+
+```
+mvn package
+tar -C /tmp -xf distribution/target/apache-druid-0.21.0-bin.tar.gz
+cd /tmp/apache-druid-0.21.0
+```
+
+
+Edit `conf/druid/single-server/micro-quickstart/_common/common.runtime.properties` to enable
+the emitter (see `Configuration` section above). Set the following properties:
+```
+otel.traces.exporter=zipkin
+otel.service.name=druid
+```
+[More about OpenTelemetry SDK Autoconfigure](https://github.com/open-telemetry/opentelemetry-java/tree/main/sdk-extensions/autoconfigure)
+
+Start the quickstart:
+
+```
+bin/start-micro-quickstart
+```
+
+Load sampel data - [example](https://druid.apache.org/docs/latest/tutorials/index.html#step-4-load-data)
+
+Then you can send queries using `curl` and a query file `query.json`:
+```
+curl -XPOST -H'Content-Type: application/json' http://localhost:8888/druid/v2/sql/ -d @query.json
+```
+
+To test propagation run simple server which read query from stdin:
```

Comment:
rather than pasting a bunch of code here, it would be nicer if we could use some existing opentelemetry collector docker images to demo how to test this.

Reply:
Yeah, absolutely
But we want to put trace context to json and then make request to druid, so I think it's not a common case
Or do you mean to create our own docker image?

Reply:
We can create our own docker image, and imply provide `make` command line in `README.md`, you can checkout this example here in golang: https://github.com/confluentinc/cc-otel-collector/tree/master/example We run the test by `make run-example-client`.